python/paddle/v2/framework/framework.py

```diff
@@ -256,7 +256,8 @@ def __init__(self,
                     self.desc.set_block_attr(attr_name, attrs[attr_name].desc)
 
         self.desc.check_attrs()
-        self.desc.infer_shape(self.block.desc)
+        if type not in {'feed', 'fetch'}:
+            self.desc.infer_shape(self.block.desc)
```

Comment:
why we can't infer shape a feed/fetch?

Reply:
InferVarType is also needed.
```
self.desc.infer_var_type(self.block.desc)
```

Reply:
here `infer_shape` is a compile-time method. Feed/Fetch op is a run-time operator, do not need to infer shape.

Reply:
> InferVarType is also needed.

Add in rest PRs